engine/player/unique_gear_shadowlands.cpp

```diff
@@ -4280,80 +4273,178 @@ void yasahm_the_riftbreaker( special_effect_t& effect )
+      swap_stat_compensation->set_duration( 0_s )->set_can_cancel( false );
+    }
+
+    if ( p->antumbra.swap )
+    {
+      swap_stat_compensation->stats.clear();
+      if ( p->antumbra.int_diff != 0.0 )
+        swap_stat_compensation->add_stat( STAT_INTELLECT, p->antumbra.int_diff );
+      if ( p->antumbra.crit_diff != 0.0 )
+        swap_stat_compensation->add_stat( STAT_CRIT_RATING, p->antumbra.crit_diff );
+      if ( p->antumbra.haste_diff != 0.0 )
+        swap_stat_compensation->add_stat( STAT_HASTE_RATING, p->antumbra.haste_diff );
+      if ( p->antumbra.vers_diff != 0.0 )
+        swap_stat_compensation->add_stat( STAT_VERSATILITY_RATING, p->antumbra.vers_diff );
+      if ( p->antumbra.mastery_diff != 0.0 )
+        swap_stat_compensation->add_stat( STAT_MASTERY_RATING, p->antumbra.mastery_diff );
+    }
+
+    set_expire_at_max_stack( true );
+    set_stack_change_callback( [ this ]( buff_t* b, int, int ) {
+      if ( b->at_max_stacks() )
+      {
+        lockout->trigger();
+        supremacy_buff->trigger();
+      }
+    } );
+  }
+};
+
 void singularity_supreme( special_effect_t& effect )
 {
-  auto lockout = make_buff( effect.player, "singularity_supreme_lockout", effect.player->find_spell( 368865 ) )
-    ->set_quiet( true );
+  /*auto lockout =
+      make_buff( effect.player, "singularity_supreme_lockout", effect.player->find_spell( 368865 ) )->set_quiet( true );
 
   auto buff =
-      make_buff<stat_buff_t>( effect.player, "singularity_supreme", effect.player->find_spell( 368863 ), effect.item );
+      make_buff<stat_buff_t>( effect.player, "singularity_supreme", effect.player->find_spell( 368863 ), effect.item );*/
 
   // despite spell data proc flags, logs seem to show it only procs on damage spell casts
   effect.proc_flags2_ = PF2_CAST_DAMAGE;
-  effect.custom_buff =
-      make_buff<stat_buff_t>( effect.player, "singularity_supreme_counter", effect.player->find_spell( 368845 ), effect.item )
-          ->set_expire_at_max_stack( true )
-          ->set_stack_change_callback( [ lockout, buff ]( buff_t* b, int, int ) {
-            if ( b->at_max_stacks() )
-            {
-              lockout->trigger();
-              buff->trigger();
-            }
-          } );
+
+  auto singularity_buff = new singularity_supreme_t(effect.player, effect.item);
+
+  effect.custom_buff = singularity_buff;
 
   new dbc_proc_callback_t( effect.player, effect );
 
   effect.player->callbacks.register_callback_trigger_function(
       effect.driver()->id(), dbc_proc_callback_t::trigger_fn_type::CONDITION,
-      [ lockout ]( const dbc_proc_callback_t*, action_t*, action_state_t* ) {
-        return !lockout->check();
+      [ singularity_buff ]( const dbc_proc_callback_t*, action_t*, action_state_t* ) {
+        return !singularity_buff->swap_stat_compensation->check() && !singularity_buff->lockout->check();
       } );
 }
 
+// Action to trigger the chain break for the Chains of Domination trinket
```

Comment:
Because I clangformatted and made a mess, here's the start of the swap action

Reply:
Noticed comment was wrong. Fixed comment,  but it's still in the right place.
Fixed one:
https://github.com/simulationcraft/simc/blob/a2b7467b50ed603c03c044c12e038b883a5e2884/engine/player/unique_gear_shadowlands.cpp#L4398